src/CalculatorUI/App.xaml.cs

```diff
@@ -0,0 +1,516 @@
+                    if (!m_preLaunched)
+                    {
+                        var newCoreAppView = CoreApplication.CreateNewView();
+                        _ = newCoreAppView.Dispatcher.RunAsync(
+                            CoreDispatcherPriority.Normal, new DispatchedHandler(async () =>
+                            {
+                                if (weak.IsAlive)
+                                {
+                                    var that = weak.Target as App;
+                                    var newRootFrame = App.CreateFrame();
+
+                                    SetMinWindowSizeAndActivate(newRootFrame, minWindowSize);
+
+                                    if (!rootFrame.Navigate(typeof(MainPage), argument))
+                                    {
+                                        // We couldn't navigate to the main page, kill the app so we have a good
+                                        // stack to debug
+                                        throw new Exception();
+                                    }
+
+                                    var frameService = WindowFrameService.CreateNewWindowFrameService(rootFrame, true, weak);
+                                    that.AddWindowToMap(frameService);
+
+                                    var dispatcher = CoreWindow.GetForCurrentThread().Dispatcher;
+                                    var safeFrameServiceCreation = new SafeFrameWindowCreation(frameService, that);
+                                    int newWindowId = ApplicationView.GetApplicationViewIdForWindow(CoreWindow.GetForCurrentThread());
+
+                                    ActivationViewSwitcher activationViewSwitcher = null;
+                                    var activateEventArgs = (args as IViewSwitcherProvider);
+                                    if (activateEventArgs != null)
+                                    {
+                                        activationViewSwitcher = activateEventArgs.ViewSwitcher;
+                                    }
+
+                                    if (activationViewSwitcher != null)
+                                    {
+                                        await activationViewSwitcher.ShowAsStandaloneAsync(newWindowId, ViewSizePreference.Default);
+                                        safeFrameServiceCreation.SetOperationSuccess(true);
+                                    }
+                                    else
+                                    {
+                                        var activatedEventArgs = (args as IApplicationViewActivatedEventArgs);
+                                        if ((activatedEventArgs != null) && (activatedEventArgs.CurrentlyShownApplicationViewId != 0))
+                                        {
+                                            var viewShown = await ApplicationViewSwitcher.TryShowAsStandaloneAsync(
```

Comment:
Task.Run() and ContinueWith

Reply:
SafeFrameServiceCreate problem requires here to using await statements to translate the code.